src/main/java/org/json/JSONObject.java

```diff
@@ -2629,8 +2632,34 @@ public static Object wrap(Object object) {
         return wrap(object, null);
     }
 
+    // Custom isFiniteNumber method
+    public static boolean isFiniteNumber(Object value) {
```

Comment:
This method is sub-optimal, and redundant of `testValidity` and `numberIsFinite` methods that are already available. Please use the existing function(s)